src/test/java/software/amazon/awssdk/crt/test/S3ClientTest.java

```diff
@@ -353,6 +356,180 @@ public long getLength() {
+
+            try (S3MetaRequest metaRequest = client.makeMetaRequest(metaRequestOptions)) {
+                Assert.assertEquals(Integer.valueOf(0), onPutFinishedFuture.get());
+            }
+
+            // Get request!
+
+            HttpHeader[] getHeaders = { new HttpHeader("Host", ENDPOINT),};
+
+            HttpRequest httpGetRequest = new HttpRequest("GET", "/java_round_trip_test_fc.txt", getHeaders, null);
+
+            CompletableFuture<Integer> onGetFinishedFuture = new CompletableFuture<>();
+            S3MetaRequestResponseHandler getResponseHandler = new S3MetaRequestResponseHandler() {
+
+                @Override
+                public int onResponseBody(ByteBuffer bodyBytesIn, long objectRangeStart, long objectRangeEnd) {
+                    Log.log(Log.LogLevel.Info, Log.LogSubject.JavaCrtS3, "Body Response: " + bodyBytesIn.toString());
+                    return 0;
+                }
+
+                @Override
+                public void onFinished(S3ResponseContext context) {
+                    Log.log(Log.LogLevel.Info, Log.LogSubject.JavaCrtS3,
+                            "Meta request finished with error code " + context.errorCode);
+                    if (context.errorCode != 0) {
+                        onGetFinishedFuture.completeExceptionally(
+                                new CrtS3RuntimeException(context.errorCode, context.responseStatus, context.errorPayload));
+                        return;
+                    }
+                    if(!context.didValidateChecksum) {
+                        onGetFinishedFuture.completeExceptionally(
+                                new CrtS3RuntimeException(context.errorCode, context.responseStatus, context.errorPayload));
+                        return;
+                    }
+                    if(context.checksumAlgorithm != S3ChecksumAlgorithm.AWS_SCA_CRC32) {
+                        onGetFinishedFuture.completeExceptionally(
+                                new CrtS3RuntimeException(context.errorCode, context.responseStatus, context.errorPayload));
+                        return;
+                    }
+                    onGetFinishedFuture.complete(Integer.valueOf(context.errorCode));
+                }
+            };
+
+            S3MetaRequestOptions getRequestOptions = new S3MetaRequestOptions()
+                    .withMetaRequestType(MetaRequestType.GET_OBJECT).withHttpRequest(httpGetRequest)
+                    .withResponseHandler(getResponseHandler)
+                    .withValidateChecksum(true);
+            
+            try (S3MetaRequest metaRequest = client.makeMetaRequest(getRequestOptions)) {
+                Assert.assertEquals(Integer.valueOf(0), onGetFinishedFuture.get());
+            }
+        } catch (InterruptedException | ExecutionException ex) {
+            Assert.fail(ex.getMessage());
+        }
+    }
+
+
+    @Test
+    public void testS3GetChecksums() {
```

Comment:
Do you have any tests that exercise the failure path, where validate-checksums fails?

Reply:
No, and there isn't an obvious way that jumps out at me, need to think about this.